src/main/scala/nl/knaw/dans/easy/fedora2vault/EasyFedora2vaultApp.scala

```diff
@@ -15,10 +15,68 @@
  */
 package nl.knaw.dans.easy.fedora2vault
 
-import scala.util.{ Success, Try }
+import java.io.InputStream
 
-class EasyFedora2vaultApp(configuration: Configuration)  {
+import better.files.File
+import com.yourmediashelf.fedora.client.FedoraClient
+import nl.knaw.dans.bag.v0.DansV0Bag
+import nl.knaw.dans.easy.fedora2vault.Command.FeedBackMessage
+import nl.knaw.dans.easy.fedora2vault.FoXml.{ getEmd, _ }
+import resource.{ ManagedResource, managed }
 
+import scala.util.{ Failure, Success, Try }
+import scala.xml.{ Node, XML }
 
+class EasyFedora2vaultApp(configuration: Configuration) {
+  private lazy val fedoraClient = new FedoraClient(configuration.fedoraCredentials)
 
+  def simpleTransform(datasetId: DatasetId, outputDir: File): Try[FeedBackMessage] = {
+    for {
+      foXml <- getFoXmlInputStream(datasetId).map(XML.load).tried
+      depositor <- getOwner(foXml)
+      bag <- DansV0Bag.empty(outputDir).map(_.withEasyUserAccount(depositor))
+      emd <- getEmd(foXml)
+      _ <- bag.addMetadata(emd, "emd.xml")
+      _ <- getDdm(foXml)
+        .map(bag.addMetadata(_, "ddm.xml")).getOrElse(Success(())) // TODO EASY-2683
+      _ <- getAgreementsXml(foXml)
+        .map(bag.addMetadata(_, "agreements.xml")).getOrElse(Success(())) // TODO ?
+      _ <- getMessageFromDepositor(foXml)
+        .map(bag.addMetadata(_, "message-from-depositor.txt")).getOrElse(Success(())) // TODO EMD/other/remark?
+      _ <- getFilesXml(foXml)
+        .map(bag.addMetadata(_, "files.xml")).getOrElse(Success(())) // TODO EASY-2678
+      _ <- getAditionalLicense(foXml)
+        .map(addMetadata(bag, "ADDITIONAL_LICENSE")).getOrElse(Success(()))
+      _ <- getDatasetLicense(foXml)
+        .map(addMetadata(bag, "DATASET_LICENSE")).getOrElse(Success(()))
+      _ <- bag.save()
+      _ = getManifest(foXml).map(compareManifest(bag))
+    } yield "???" // TODO what?
+  }
+
+  protected def getFoXmlInputStream(datasetId: DatasetId): ManagedResource[InputStream] = {
+    // copy of https://github.com/DANS-KNAW/easy-export-dataset/blob/6e656c6e6dad19bdea70694d63ce929ab7b0ad2b/src/main/scala/nl.knaw.dans.easy.export/FedoraProvider.scala#L55-L58
+    managed(FedoraClient.getObjectXML(datasetId).execute(fedoraClient))
+      .flatMap(response => managed(response.getEntityInputStream))
+  }
```

Comment:
a quick side note on class structure:

currently you have a this `EasyFedora2vaultApp` class as well as a `FoXml` object containing all sorts of functions for extractions from your `foxml`.
Having this function here seems a bit odd to me; I would expect this to go in the `FoXml` class.

Why don't you make it as follows (this is a sketch, you can develop it some more yourself):

```scala
class FoXml(xml: Node) {
  def getEmd: Try[Node] = getStream("EMD", "easymetadata")
  def getAmd: Try[Node] = getStream("AMD", "administrative-md")
  // etcetera
}
object FoXml {
  def apply(datasetId: DatasetId)(implicit fedoraClient: FedoraClient): Try[FoXml] = {
    managed(FedoraClient.getObjectXML(datasetId).execute(fedoraClient))
      .flatMap(response => managed(response.getEntityInputStream))
      .map(XML.load)
      .tried
      .map(new FoXml(_))
  }
}
```

Reply:
Just the wrong name. The same method is used for other objects with their own fedora-id like the licenses and later for payload files. I need to have the snippet isolated for testing, which is not yet properly implemented for the licenses. I might need some variant of FedoraProvider as in easy-export-dataset.

It is good to have your other comments in this early stage, it will help to focus properly.